src/app.rs

```diff
@@ -30,7 +31,12 @@ fn load<T: DeserializeOwned>(cc: &eframe::CreationContext<'_>, key: &'static str
     }
 }
 
-#[derive(Debug, Serialize, Deserialize)]
```

Comment:
This change (and related ones) seems to break the web app. Here is the compiler error I get:
```
error[E0277]: the trait bound `SolverInput: Serialize` is not satisfied
   --> src/app.rs:98:14
    |
98  |             .spawn(concat!("./webworker", env!("RANDOM_SUFFIX"), ".js"))
    |              ^^^^^ the trait `Serialize` is not implemented for `SolverInput`
    |
    = note: for local types consider adding `#[derive(serde::Serialize)]` to your `SolverInput` type
    = note: for types from other crates check whether the crate offers a `serde` feature flag
[...]
```
I tested simply adding `Serialize` and `Deserialize` (back) to the `derive` of stuff until it compiled, however that just resulted in the button not doing anything in the web version. (I'm not even sure why it allows me to do that, for native having it attempt to compile it, results in a compiler error about not being able to derive `Serialize`/`Deserialize` for `Arc<AtomicBool>`.)
So it seems like the `AtomicFlag` can not be shared as easily for the web version.